Porting.md

```diff
@@ -0,0 +1,145 @@
+# Porting Guide
+
+## Ports' Status
+The status of ports is kept in the main module [README](README.md).
+
+## Overview
+ZikiChombo sio code has a porting process that is unconventional due to the
+nature of sound support on host systems.  Usually, to port a system, on takes some
+desired cross-host functionality, such as a filesystem, and implements it with
+the OS or OS/hardware combination of interest and that's that (to make a long
+story short).
+
+For sound, the problem is that hosts usually have multiple different software
+components as entry points to playing and capturing sound, and moreover these
+entry points have some mutually exclusive properties, such as lower bounds on
+latency, availability of duplex synchronised by hardware audio clock, ability
+to be shared accross many users or applications, etc.
+
+Although the main CPU hardware architectures can certainly be a concern, it has
+not been so far in our experience.  So we focus on organizing around host,
+which is taken to be the value of runtime.GOOS in a Go program.
+
+# Hosts
+A host often has a hardware abstraction layer, hardware drivers, and some
+levels of interacting with these layers to coordinate and make secure demands
+related to playing and capturing sound on a device.
+
+To simplify the end use case, we want to make all this transparent to the
+consumer of sio, and let them simply work with sound.{Source,Sink,Duplex}.
+
+For each Host, ZikiChombo defines a default entry point to accomplish this.
+
+All entry points are available both to the consumer of ZikiChombo and either
+with ZikiChombo or as 3rd party implementations.  So if you're working with
+VoIP and have precise duplex echo cancellation needs, or a music app that
+listens and plays in real time with feedback, or writing a PulseAudio
+replacement, you're more likely to be interested in using a specialized entry
+point than the default.
+
+There is a directory ZikiChombo/sio/ports/{runtime.GOOS} for each host. 
+
+# Entry Points
+An entry point defines a proper subset of the functionality listed in the main
```

Comment:
Re: *proper subset*. Are there no entry points which may implement all the functionality listed in the main README? (and as such, not be proper subsets, but rather subsets).

Reply:
meant to say non-empty subset which is to my understanding not a proper subset... 

but I just dropped "proper" alltogether, I think any reader actually thinking about porting something would likely already have in mind a subset of interest.